graph/graph.go

```diff
@@ -245,11 +250,6 @@ func (graph *Graph) Register(img *image.Image, layerData archive.Reader) (err er
 		}
 	}()
 
-	// (This is a convenience to save time. Race conditions are taken care of by os.Rename)
-	if graph.Exists(img.ID) {
-		return fmt.Errorf("Image %s already exists", img.ID)
```

Comment:
Interesting. This may the root cause of the pull corruption. Above, we check for `nil` against the error and remove if there is an error. If the image exists, we effectively remove it.

I'd like to see if #9718 gets addressed with this fix.